mono/mini/llvm-jit.cpp

```diff
@@ -297,12 +297,13 @@ class MonoLLVMJIT {
 		initializeScalarOpts(registry);
 		initializeInstCombine(registry);
 		initializeTarget(registry);
+		initializeLoopIdiomRecognizeLegacyPassPass(registry);
 		linkCoreCLRGC(); // Mono uses built-in "coreclr" GCStrategy
 
 		// FIXME: find optimal mono specific order of passes
 		// see https://llvm.org/docs/Frontend/PerformanceTips.html#pass-ordering
 		// the following order is based on a stripped version of "OPT -O2"
-		const char *default_opts = " -simplifycfg -sroa -lower-expect -instcombine -licm -simplifycfg -lcssa -indvars -loop-deletion -gvn -memcpyopt -sccp -bdce -instcombine -dse -simplifycfg -enable-implicit-null-checks";
+		const char *default_opts = " -simplifycfg -sroa -lower-expect -instcombine -loop-rotate -licm -simplifycfg -lcssa -loop-idiom -indvars -loop-deletion -gvn -memcpyopt -sccp -bdce -instcombine -dse -simplifycfg -enable-implicit-null-checks";
 		const char *opts = g_getenv ("MONO_LLVM_OPT");
```

Comment:
`-loop-idiom` is the pass that recognizes memset in a loop (also needs `-loop-rotate`). So now both AOT and JIT can do it.

Reply:
![image](https://user-images.githubusercontent.com/523221/69159582-31fcb180-0af9-11ea-9247-28b6755a35e4.png)